Run started:2024-02-10 22:00:14.593206

Test results:
>> Issue: [B110:try_except_pass] Try, Except, Pass detected.
   Severity: Low   Confidence: High
   CWE: CWE-703 (https://cwe.mitre.org/data/definitions/703.html)
   More Info: https://bandit.readthedocs.io/en/1.7.7/plugins/b110_try_except_pass.html
   Location: .\pystemon.py:243:16
242	                    logger.info("{}: queue size={}".format(repr(site), site.queue.qsize()))
243	                except:
244	                    pass
245	        except PystemonConfigException as e:

--------------------------------------------------
>> Issue: [B110:try_except_pass] Try, Except, Pass detected.
   Severity: Low   Confidence: High
   CWE: CWE-703 (https://cwe.mitre.org/data/definitions/703.html)
   More Info: https://bandit.readthedocs.io/en/1.7.7/plugins/b110_try_except_pass.html
   Location: .\pystemon\config.py:108:8
107	            res = self._configfile == other._configfile
108	        except Exception as e:
109	            pass
110	        return res

--------------------------------------------------
>> Issue: [B506:yaml_load] Use of unsafe yaml load. Allows instantiation of arbitrary objects. Consider yaml.safe_load().
   Severity: Medium   Confidence: High
   CWE: CWE-20 (https://cwe.mitre.org/data/definitions/20.html)
   More Info: https://bandit.readthedocs.io/en/1.7.7/plugins/b506_yaml_load.html
   Location: .\pystemon\config.py:307:29
306	                # only for 5.1+
307	                yamlconfig = yaml.load(open(configfile), Loader=yaml.FullLoader)
308	            else:

--------------------------------------------------
>> Issue: [B506:yaml_load] Use of unsafe yaml load. Allows instantiation of arbitrary objects. Consider yaml.safe_load().
   Severity: Medium   Confidence: High
   CWE: CWE-20 (https://cwe.mitre.org/data/definitions/20.html)
   More Info: https://bandit.readthedocs.io/en/1.7.7/plugins/b506_yaml_load.html
   Location: .\pystemon\config.py:309:29
308	            else:
309	                yamlconfig = yaml.load(open(configfile))
310	        except yaml.YAMLError as exc:

--------------------------------------------------
>> Issue: [B506:yaml_load] Use of unsafe yaml load. Allows instantiation of arbitrary objects. Consider yaml.safe_load().
   Severity: Medium   Confidence: High
   CWE: CWE-20 (https://cwe.mitre.org/data/definitions/20.html)
   More Info: https://bandit.readthedocs.io/en/1.7.7/plugins/b506_yaml_load.html
   Location: .\pystemon\config.py:318:34
317	                logger.debug("loading include '{0}'".format(includes))
318	                yamlconfig.update(yaml.load(open(includes)))
319	            except Exception as e:

--------------------------------------------------
>> Issue: [B324:hashlib] Use of weak MD5 hash for security. Consider usedforsecurity=False
   Severity: High   Confidence: High
   CWE: CWE-327 (https://cwe.mitre.org/data/definitions/327.html)
   More Info: https://bandit.readthedocs.io/en/1.7.7/plugins/b324_hashlib.html
   Location: .\pystemon\pastie\__init__.py:85:27
84	            try:
85	                self.md5 = hashlib.md5(self.pastie_content).hexdigest()
86	                logger.debug('Pastie {site} {id} has md5: "{md5}"'.format(site=self.site.name, id=self.id, md5=self.md5))

--------------------------------------------------
>> Issue: [B307:blacklist] Use of possibly insecure function - consider using safer ast.literal_eval.
   Severity: Medium   Confidence: High
   CWE: CWE-78 (https://cwe.mitre.org/data/definitions/78.html)
   More Info: https://bandit.readthedocs.io/en/1.7.7/blacklists/blacklist_calls.html#b307-eval
   Location: .\pystemon\pastiesearch.py:9:25
8	            self.regex_flags = regex['regex-flags']
9	            self.flags = eval(self.regex_flags)
10	        else:

--------------------------------------------------
>> Issue: [B311:blacklist] Standard pseudo-random generators are not suitable for security/cryptographic purposes.
   Severity: Low   Confidence: High
   CWE: CWE-330 (https://cwe.mitre.org/data/definitions/330.html)
   More Info: https://bandit.readthedocs.io/en/1.7.7/blacklists/blacklist_calls.html#b311-random
   Location: .\pystemon\pastiesite.py:115:33
114	                while not self.kill_received:
115	                    sleep_time = random.randint(self.update_min, self.update_max)
116	                    # grabs site from queue

--------------------------------------------------
>> Issue: [B311:blacklist] Standard pseudo-random generators are not suitable for security/cryptographic purposes.
   Severity: Low   Confidence: High
   CWE: CWE-330 (https://cwe.mitre.org/data/definitions/330.html)
   More Info: https://bandit.readthedocs.io/en/1.7.7/blacklists/blacklist_calls.html#b311-random
   Location: .\pystemon\proxy.py:94:24
93	            if self.proxies_list:
94	                proxy = random.choice(tuple(self.proxies_list))
95	        return proxy

--------------------------------------------------
>> Issue: [B113:request_without_timeout] Requests call without timeout
   Severity: Medium   Confidence: Low
   CWE: CWE-400 (https://cwe.mitre.org/data/definitions/400.html)
   More Info: https://bandit.readthedocs.io/en/1.7.7/plugins/b113_request_without_timeout.html
   Location: .\pystemon\storage\telegramstorage.py:31:16
30	                logger.debug('Sending message to telegram {} for pastie_id {}'.format(url, pastie.id))
31	                requests.post(url, data={'chat_id': self.chat_id, 'text': message})
32	            except Exception as e:

--------------------------------------------------
>> Issue: [B110:try_except_pass] Try, Except, Pass detected.
   Severity: Low   Confidence: High
   CWE: CWE-703 (https://cwe.mitre.org/data/definitions/703.html)
   More Info: https://bandit.readthedocs.io/en/1.7.7/plugins/b110_try_except_pass.html
   Location: .\pystemon\throttler.py:89:12
88	                break
89	            except Exception as e:
90	                pass
91	        logger.info('ThreadThrottler[{}] exited'.format(site))

--------------------------------------------------
>> Issue: [B311:blacklist] Standard pseudo-random generators are not suitable for security/cryptographic purposes.
   Severity: Low   Confidence: High
   CWE: CWE-330 (https://cwe.mitre.org/data/definitions/330.html)
   More Info: https://bandit.readthedocs.io/en/1.7.7/blacklists/blacklist_calls.html#b311-random
   Location: .\pystemon\ua.py:83:19
82	        if self.user_agents_list:
83	            return random.choice(self.user_agents_list)
84	        return 'Python-urllib/2.7'

--------------------------------------------------

Code scanned:
	Total lines of code: 2225
	Total lines skipped (#nosec): 0
	Total potential issues skipped due to specifically being disabled (e.g., #nosec BXXX): 0

Run metrics:
	Total issues (by severity):
		Undefined: 0
		Low: 6
		Medium: 5
		High: 1
	Total issues (by confidence):
		Undefined: 0
		Low: 1
		Medium: 0
		High: 11
Files skipped (0):
